// web-audio-oscillators/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.customOscillators = exports.customOscillatorTypes = void 0;
const sine_1 = require("./wave-tables/sine");
const square_1 = require("./wave-tables/square");
const sawtooth_1 = require("./wave-tables/sawtooth");
const triangle_1 = require("./wave-tables/triangle");
const chiptune_1 = require("./wave-tables/chiptune");
const organ_1 = require("./wave-tables/organ");
const organ2_1 = require("./wave-tables/organ2");
const bass_1 = require("./wave-tables/bass");
const brass_1 = require("./wave-tables/brass");
const wave_tables_1 = require("@mohayonao/wave-tables");
const wave_tables_2 = require("@mohayonao/wave-tables");
const wave_tables_3 = require("@mohayonao/wave-tables");
const wave_tables_4 = require("@mohayonao/wave-tables");
const wave_tables_5 = require("@mohayonao/wave-tables");
const wave_tables_6 = require("@mohayonao/wave-tables");
const wave_tables_7 = require("@mohayonao/wave-tables");
const wave_tables_8 = require("@mohayonao/wave-tables");
const wave_tables_9 = require("@mohayonao/wave-tables");
const wave_tables_10 = require("@mohayonao/wave-tables");
const wave_tables_11 = require("@mohayonao/wave-tables");
const wave_tables_12 = require("@mohayonao/wave-tables");
const wave_tables_13 = require("@mohayonao/wave-tables");
const wave_tables_14 = require("@mohayonao/wave-tables");
const wave_tables_15 = require("@mohayonao/wave-tables");
const waveTables = {
    sine: sine_1.sine,
    square: square_1.square,
    square2: wave_tables_1.DroppedSquare,
    sawtooth: sawtooth_1.sawtooth,
    triangle: triangle_1.triangle,
    triangle2: wave_tables_2.WarmTriangle,
    chiptune: chiptune_1.chiptune,
    organ: organ_1.organ,
    organ2: organ2_1.organ2,
    organ3: wave_tables_3.Organ2,
    organ4: wave_tables_4.Organ3,
    organ5: wave_tables_5.Dissonant2,
    bass: bass_1.bass,
    bass2: wave_tables_6.Bass,
    bass3: wave_tables_7.BassSubDub,
    bass4: wave_tables_8.BassFuzz2,
    brass: brass_1.brass,
    brass2: wave_tables_9.Trombone,
    aah: wave_tables_10.PhonemeAh,
    ooh: wave_tables_11.PhonemeOoh,
    eeh: wave_tables_12.PhonemeEe,
    buzz: wave_tables_13.Buzzy1,
    buzz2: wave_tables_14.DynaEPBright,
    dissonance: wave_tables_15.DissonantPiano,
};
exports.customOscillatorTypes = Object.keys(waveTables);
exports.customOscillators = exports.customOscillatorTypes.reduce((obj, type) => {
    return Object.assign(obj, {
        [type]: createCustomOscillator.bind(null, waveTables[type]),
    });
}, {});
function createCustomOscillator(waveTable, context) {
    const oscillator = context.createOscillator();
    oscillator.setPeriodicWave(context.createPeriodicWave(Float32Array.from(waveTable.real), Float32Array.from(waveTable.imag)));
    return oscillator;
}
